### PR TITLE
Jennyf/logger

### PR DESCRIPTION
**Pull request by jennyf19**
New MsalLogger class to hold logger functionality. Platform specific loggers implement ILogger interface instead of abstract LoggerBase. Functionality from LoggerBase is now in MsalLogger. All references to PlatformPlugin log methods now use MsalLogger. Unit tests for LoggerCallbackTests to test callback for Error, Warning, Info and Verbose.

### CODE REVIEW

File: src/MSAL.Common/Logger.cs [hunk 1]
```diff
@@ -30,52 +30,48 @@
 
 namespace Microsoft.Identity.Client
 {
-    internal class Logger : LoggerBase
+    internal class Logger : ILogger
```
Comment (jennyf19) [inline src/MSAL.Common/Logger.cs]: Done.

Comment (RandalliLama) [inline src/MSAL.Common/Logger.cs]: Actually, this was my mistake.  :(. Anyway, the change you made makes sense.

**Comment by kpanwar** [comment]
@jennyf19 - please merge dev branch into your code to resolve conflicts.

File: src/MSAL.PCL/Internal/IMsalLogger.cs [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+    internal interface IMsalLogger
+    {
+        void Error(string message);
+        void Warning(string message);
+        void Information(string message);
+        void Verbose(string message);
+
+        void Error(Exception ex);
+        void Warning(Exception ex);
+        void Information(Exception ex);
```
Comment (kpanwar) [inline src/MSAL.PCL/Internal/IMsalLogger.cs]: why do we have information, warning, verbose method logging Exception object?

File: src/MSAL.PCL/MsalLogger.cs [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+
+        private void LogMessage(string logMessage, LogLevel logLevel)
+        {
+            if (logLevel > ApplicationLogLevel) return;
+
+            //format log message;
+            string correlationId = (CorrelationId.Equals(Guid.Empty))
+                ? "No CorrelationId"
+                : CorrelationId.ToString();
+            string log = string.Format(CultureInfo.CurrentCulture, "{0}: {1}: {2}", DateTime.UtcNow, correlationId,
```
Comment (kpanwar) [inline src/MSAL.PCL/MsalLogger.cs]: this is not the correct logging format. Please refer to ryan's doc.

File: src/MSAL.PCL/MsalLogger.cs [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+    {
+        public enum LogLevel
+        {
+            Error = 0,
+            Warning = 1,
+            Info = 2,
+            Verbose = 3
+        }
+
+        internal MsalLogger(Guid correlationId)
```
Comment (kpanwar) [inline src/MSAL.PCL/MsalLogger.cs]: log methods are missing containsPII flag.

File: src/MSAL.PCL/LoggerCallbackHandler.cs [hunk 5]
```diff
@@ -78,7 +56,7 @@ public static IMsalLogCallback Callback
             }
         }
 
-        internal static void ExecuteCallback(LogLevel level, string message)
+        internal static void ExecuteCallback(MsalLogger.LogLevel level, string message)
```
Comment (kpanwar) [inline src/MSAL.PCL/LoggerCallbackHandler.cs]: as per the spec, developer should be allowed to set callback only once. I think that change is missing in this class.

File: src/MSAL.PCL/MsalLogger.cs [hunk 6]
```diff
@@ -0,0 +1,109 @@
...
+//
+//------------------------------------------------------------------------------
+
+using System;
+using System.Globalization;
+using Microsoft.Identity.Client.Internal;
+
+namespace Microsoft.Identity.Client
+{
+    public class MsalLogger
```
Comment (kpanwar) [inline src/MSAL.PCL/MsalLogger.cs]: Rename to Logger.

File: src/MSAL.PCL/MsalLogger.cs [hunk 7]
```diff
@@ -0,0 +1,109 @@
...
+            this.CorrelationId = correlationId;
+        }
+
+        internal Guid CorrelationId { get; set; }
+
+        internal LogLevel ApplicationLogLevel { get; set; } = LogLevel.Info;
+
+        #region LogMessages
+
+        public void Error(string message)
```
Comment (kpanwar) [inline src/MSAL.PCL/MsalLogger.cs]: add documentation headers on all public methods of the public classes

**Pull request closed without merging**